typings/index.d.ts

```diff
@@ -703,13 +703,14 @@ declare module 'discord.js' {
   }
 
   export class DiscordAPIError extends Error {
-    constructor(path: string, error: unknown, method: string, httpStatus: number);
+    constructor(error: unknown, status: number, request: HTTPRequest);
```

Comment:
I don't think there is much point in documenting parts of our APIRequest class:
```suggestion
    constructor(error: unknown, status: number, request: unknown);
```